WalletWasabi.Tests/UnitTests/SmartTransactionTests.cs

```diff
@@ -82,46 +82,245 @@ public void FirstSeenBackwardsCompatibility()
+				"0200000001268171371edff285e937adeea4b37b78000c0566cbb3ad64641713ca42171bf6000000006a473044022070b2245123e6bf474d60c5b50c043d4c691a5d2435f09a34a7662a9dc251790a022001329ca9dacf280bdf30740ec0390422422c81cb45839457aeb76fc12edd95b3012102657d118d3357b8e0f4c2cd46db7b39f6d9c38d9a70abcb9b2de5dc8dbfe4ce31feffffff02d3dff505000000001976a914d0c59903c5bac2868760e90fd521a4665aa7652088ac00e1f5050000000017a9143545e6e33b832c47050f24d3eeb93c9c03948bc787b32e1300",
+				"0100000002d8c8df6a6fdd2addaf589a83d860f18b44872d13ee6ec3526b2b470d42a96d4d000000008b483045022100b31557e47191936cb14e013fb421b1860b5e4fd5d2bc5ec1938f4ffb1651dc8902202661c2920771fd29dd91cd4100cefb971269836da4914d970d333861819265ba014104c54f8ea9507f31a05ae325616e3024bd9878cb0a5dff780444002d731577be4e2e69c663ff2da922902a4454841aa1754c1b6292ad7d317150308d8cce0ad7abffffffff2ab3fa4f68a512266134085d3260b94d3b6cfd351450cff021c045a69ba120b2000000008b4830450220230110bc99ef311f1f8bda9d0d968bfe5dfa4af171adbef9ef71678d658823bf022100f956d4fcfa0995a578d84e7e913f9bb1cf5b5be1440bcede07bce9cd5b38115d014104c6ec27cffce0823c3fecb162dbd576c88dd7cda0b7b32b0961188a392b488c94ca174d833ee6a9b71c0996620ae71e799fc7c77901db147fa7d97732e49c8226ffffffff02c0175302000000001976a914a3d89c53bb956f08917b44d113c6b2bcbe0c29b788acc01c3d09000000001976a91408338e1d5e26db3fce21b011795b1c3c8a5a5d0788ac00000000"
+			};
+			var defaultTx = Transaction.Parse(txHexes.First(), defaultNetwork);
+
+			var heights = new List<Height>
+			{
+				Height.Unknown,
+				Height.Mempool,
+				new Height(0),
+				new Height(100),
+				new Height(int.MaxValue)
+			};
+			var defaultHeight = new Height(0);
+
+			var blockHashes = new List<uint256>
+			{
+				null,
+				uint256.Parse("000000000000000000093e2e41b170cd9e10ed8a0469c9719abd227d5226672f")
+			};
+
+			var blockIndexes = new List<int>
+			{
+				0,
+				1,
+				100,
+				int.MaxValue
+			};
+
+			var labels = new List<string>
+			{
+				"",
+				" ",
+				"foo",
+				"foo, bar",
+				"               :foo:bar:buz: ",
+				"~!@#$%^&*()"
+			};
+
+			var firstSeens = new List<DateTimeOffset>
+			{
+				DateTimeOffset.UtcNow,
+				DateTimeOffset.Now,
+				DateTimeOffset.MaxValue,
+				DateTimeOffset.MinValue,
+				DateTimeOffset.UnixEpoch
+			};
+
+			var isReplacements = new List<bool>
+			{
+				false,
+				true
+			};
+
+			foreach (var network in networks)
+			{
+				foreach (var txHex in txHexes)
+				{
+					var tx = Transaction.Parse(txHex, network);
+					foreach (var height in heights)
+					{
+						yield return new object[] { new SmartTransaction(tx, height), network };
+					}
+				}
+			}
+
+			foreach (var blockHash in blockHashes)
+			{
+				yield return new object[] { new SmartTransaction(defaultTx, defaultHeight, blockHash), defaultNetwork };
+			}
+
+			foreach (var blockIndex in blockIndexes)
+			{
+				yield return new object[] { new SmartTransaction(defaultTx, defaultHeight, blockIndex: blockIndex), defaultNetwork };
+			}
+			foreach (var label in labels)
```

Comment:
Add separation line.
```suggestion

			foreach (var label in labels)
```